micrometer/src/main/java/feign/micrometer/MeteredClient.java

```diff
@@ -45,38 +45,82 @@ public MeteredClient(Client client, MeterRegistry meterRegistry, MetricName metr
   @Override
   public Response execute(Request request, Options options) throws IOException {
     final RequestTemplate template = request.requestTemplate();
-
+    final Timer.Sample sample = Timer.start(meterRegistry);
     try {
-      return meterRegistry.timer(
-          metricName.name(),
-          metricName.tag(template.methodMetadata(), template.feignTarget()))
-          .recordCallable(() -> {
-            Response response = client.execute(request, options);
-            meterRegistry.counter(
-                metricName.name("http_response_code"),
-                metricName.tag(
-                    template.methodMetadata(),
-                    template.feignTarget(),
-                    Tag.of("http_status", String.valueOf(response.status())),
-                    Tag.of("status_group", response.status() / 100 + "xx")))
-                .increment();
-            return response;
-          });
+      final Response response = client.execute(request, options);
+      countSuccessResponseCode(template, options, response.status());
+      final Timer timer = createSuccessTimer(template, options);
+      sample.stop(timer);
+      return response;
     } catch (FeignException e) {
-      meterRegistry.counter(
-          metricName.name("http_response_code"),
-          metricName.tag(
-              template.methodMetadata(),
-              template.feignTarget(),
-              Tag.of("http_status", String.valueOf(e.status())),
-              Tag.of("status_group", e.status() / 100 + "xx")))
-          .increment();
+      countErrorResponseCode(template, options, e.status(), e);
       throw e;
     } catch (IOException | RuntimeException e) {
+      sample.stop(createExceptionTimer(template, options, e));
       throw e;
     } catch (Exception e) {
+      sample.stop(createExceptionTimer(template, options, e));
       throw new IOException(e);
     }
   }
 
+  private void countSuccessResponseCode(RequestTemplate template,
+                                        Options options,
+                                        int responseStatus) {
+    final List<Tag> tags = extraSuccessTags(template, options);
```

Comment:
I think code would be a bit better to read if "extraTags" returned "Tag[]"

```
metricName.tag(template.methodMetadata(), template.feignTarget())
  .and(extraTags(...))
  .and("xyz", 10);
```

Reply:
Done.